### PR TITLE
Add MatchAttributeWithValue condition

### PR DESCRIPTION
**Pull request by dstekanov**
## Proposed changes

If resolves a feature request #996

## Checklist
- [x] Checkstyle and unit tests pass locally with my changes by running `gradlew check chrome_headless firefox_headless` command
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] I have added necessary documentation (if appropriate)

### CODE REVIEW

File: src/main/java/com/codeborne/selenide/Condition.java [hunk 1]
```diff
@@ -117,6 +118,18 @@ public static Condition attribute(String attributeName, String expectedAttribute
     return new AttributeWithValue(attributeName, expectedAttributeValue);
   }
 
+  /**
+   * Assert that given element's attribute matches given regular expression
+   *
+   * <p>Sample: <code>$("h1").should(attributeMatching(".*12345.*"))</code></p>
```
Comment (asolntsev) [inline src/main/java/com/codeborne/selenide/Condition.java]: The Sample is invalid: `attributeMatching` has 2 parameters: `name` and `value`.

File: src/main/java/com/codeborne/selenide/Condition.java [hunk 2]
```diff
@@ -262,7 +275,7 @@ public static Condition textCaseSensitive(String text) {
   }
 
   /**
-   * Assert that element is exactly (case insensitive) given text 
+   * Assert that element is exactly (case insensitive) given text
```
Comment (asolntsev) [inline src/main/java/com/codeborne/selenide/Condition.java]: "is" -> "has"

File: src/main/java/com/codeborne/selenide/conditions/MatchAttributeWithValue.java [hunk 3]
```diff
@@ -0,0 +1,40 @@
...
+import static java.util.regex.Pattern.DOTALL;
+
+public class MatchAttributeWithValue extends Condition {
+  private final String attributeName;
+  private final Pattern attributeRegex;
+
+  public MatchAttributeWithValue(String attributeName, String attributeRegex) {
+    super("match attribute");
+    this.attributeName = attributeName;
+    this.attributeRegex = Pattern.compile(attributeRegex, DOTALL);
```
Comment (asolntsev) [inline src/main/java/com/codeborne/selenide/conditions/MatchAttributeWithValue.java]: Why DOTALL?
As I know, it's needed to support multiline Strings, which hardly can happen to attribute values.

File: src/main/java/com/codeborne/selenide/Condition.java (hunk 1 above)
Comment (dstekanov) [reply src/main/java/com/codeborne/selenide/Condition.java]: thanks, done!

File: src/main/java/com/codeborne/selenide/conditions/MatchAttributeWithValue.java (hunk 3 above)
Comment (dstekanov) [reply src/main/java/com/codeborne/selenide/conditions/MatchAttributeWithValue.java]: Thanks for review Andrei! 
You are right! I removed the flag, seems this is not needed there (at least, I did not find appropriate)

**Pull request merged**